frontend/src/components/common/form/StepForm/SteppedForm.tsx

```diff
@@ -0,0 +1,83 @@
+import Stepper from 'components/common/Stepper';
+import { Form, Formik, FormikConfig, getIn, useFormikContext } from 'formik';
+import * as React from 'react';
+import { StepperFormProvider } from './context';
+import { IStepperFieldProps, ISteppedFormProps, ISteppedFormValues } from './types';
+
+export const StepperField: React.FC<IStepperFieldProps> = ({ name, steps }) => {
+  const { values, setFieldValue } = useFormikContext();
+  return (
+    <Stepper
+      steps={steps}
+      activeStep={getIn(values, name)}
+      onChange={(step, index) => setFieldValue(name, index)}
+    />
+  );
+};
+
+/**
+ * A formik form with a steppper
+ * @component
+ * @example
+ * const StepperFormContent: React.FC = () => {
+ * const stepper = useFormStepper();
+ *   return (
+ *    <>
+ *      <p>Step {stepper.current}</p>
+ *      <Field name="data.name"/>
+ *      <Button onClick={() => stepper.goBack()}>Previous<Button/>
+ *      <Button onClick={() => stepper.gotoNext()}>Next<Button/>
+ *      <Button type='submit'>Save<Button/>
+ *    </>
+ *   )
+ * }
+ * 
+ * interface IBuilding {
+ *  id?: number;
+ *  name: string;
+ * }
+ * 
+ * const BuildingForm = () => {
+ *  return (
+*          <SteppedForm<IBuilding>
+            steps={[
+              { route: 'building-id', title: 'Building ID', completed: false, canGoToStep: true },
```

Comment:
wondering about canGoToStep, is the idea that you update this as you complete steps, or is this a static list? I'm not clear on the design here, can you skip steps or can you only move ahead one step at a time?